init-tools.sh

```diff
@@ -121,7 +121,13 @@ if [ ! -e $__DOTNET_PATH ]; then
     __DOTNET_LOCATION="https://dotnetcli.azureedge.net/dotnet/Sdk/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.tar.gz"
 
     install_dotnet_cli() {
-        echo "Installing '${__DOTNET_LOCATION}' to '$__DOTNET_PATH/dotnet.tar'" >> "$__init_tools_log"
+        if [[ ! "$DotNetBootstrapCliTarPath" = "" ]]; then
+            echo "Running 'tar -xf $DotNetBootstrapCliTarPath' in '$__DOTNET_PATH'"
```

Comment:
Why not just capture the tarball path in a variable and if DotNetBootstrapCliTarPath is set then set it to that otherwise download it and set it to that. Then you can share the cd and tar commands across both. Something like:

```
rm -rf -- "$__DOTNET_PATH/*"
$DotnetTarballPath = $DotNetBootstrapCliTarPath

if [[  "$DotnetTarballPath" = "" ]]; then
         echo "Downloading '${__DOTNET_LOCATION}' to '$__DOTNET_PATH/dotnet.tar'"
         # curl has HTTPS CA trust-issues less often than wget, so lets try that first.
         if command -v curl > /dev/null; then
             curl --retry 10 -sSL --create-dirs -o $__DOTNET_PATH/dotnet.tar ${__DOTNET_LOCATION}
         else
             wget -q -O $__DOTNET_PATH/dotnet.tar ${__DOTNET_LOCATION}
         fi
         $DotNetTarballPath = $__DOTNET_PATH/dotnet.tar 
fi 

echo "Installing '$DotNetTarballPath` into '$__DOTNET_PATH'"
cd $__DOTNET_PATH
tar -xf $DotNetTarballPath
```

Reply:
Sounds like you prefer that code.  I can update.

Reply:
Done